### PR TITLE
feat: new function for gap-filling

### PR DESCRIPTION
**Pull request by Hao-Chalmers**
### Main improvements in this PR:
* a gap-filling function, `gapfill4EssentialTasks`, was developed for introducing reactions from the reference model into an input model, so that the resulting model can undertake all essential tasks, including growth on Ham's media. The changes introduced by this function are recorded in `modelChanges` variable for manual inspection. By default, the biomass objective equation is set to "biomass_components" that is constituted by generic components occurring in an eukaryotic cell.

**I hereby confirm that I have:**

- [X] Tested my code on my own computer for running the model
- [X] Selected `devel` as a target branch

### CODE REVIEW

**Review by JonathanRob — APPROVED** [review]
Very nice addition. I tested this on a few examples by randomly deleting a subset of reactions, and it worked fine.

File: code/gapfill4EssentialTasks.m [hunk 1]
```diff
@@ -102,6 +102,8 @@
 modelChanges = docModelChanges(model_orig, outModel);
 
 
+end
```
Comment (JonathanRob) [inline code/gapfill4EssentialTasks.m:105]: This doesn't change anything, but I believe that sub-functions should come _after_ the main function `end` statement, so I have corrected that here.

Comment (Hao-Chalmers) [reply code/gapfill4EssentialTasks.m:105]: Now this sub-function just blocks the other biomass reactions. I wonder if it's better to just delete them?

Comment (JonathanRob) [reply code/gapfill4EssentialTasks.m:105]: I don't think it makes a difference, so maybe just leave it as is for now.

Comment (Hao-Chalmers) [reply code/gapfill4EssentialTasks.m:105]: ok

**Pull request merged**